src/nvim/drawline.c

```diff
@@ -2632,7 +2638,7 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, int col_rows, s
         && !wp->w_p_wrap
         && wlv.filler_todo <= 0
         && wlv.col == grid->cols - 1
-        && !has_fold) {
+        && !has_foldtext) {
```

Comment:
You need to add conditions below so that a folded line behaves like a non-wrapped line.